src/jquery.eclipsefdn-api.js

```diff
@@ -1949,10 +1952,134 @@
+        $promosContainer.on("fetchPageItemsAd", retrievePromos);
+      }
+      // trigger first update of promos using first page
+      retrievePromosByPage($promosContainer, 1, 10);
+
+      /**
+       * Listener callback method for fetchPageItemsAd event.
+       */
+      function retrievePromos(event, page, pageSize) {
+        retrievePromosByPage(event.target, page, pageSize);
+      }
+
+      function retrievePromosByPage(contextEl, page, pageSize) {
+        var $currentContainer = $(contextEl);
+        var $parent = $currentContainer.parent();
+        var displayCount = $parent.data("count") || pageSize || 10;
+
+        // generate the URL for retrieve promotions
+        var url = self.settings.adsUrl;
+        var filter = "?page=" + page;
+        filter += "&pagesize=" + displayCount;
+        filter += convertDataToURLParameters($parent, "publish-target", "publish_to", undefined);
+        
+        console.log(url + filter);
+        // retrieve the promo data via ajax
+        $.ajax(url + filter, {
+          dataType: "json",
+          type: "GET",
+          success: function(data) {
+            if (data["ads"] === undefined) {
+              console.log("Could not load promotional content. AD-01");
+            }
+            // add the index of the ad for printing out index to all ads page
+            for (var i = 0; i < data["ads"].length; i++) {
+              data["ads"][i].idx = i;
+            }
+            // call and write the actual promo content
+            writePromoContent($currentContainer, data["ads"], self.settings);
+
+            // add the pagination bar 
+            if ($parent.data("pagination") === true && $parent.find("nav").length === 0) {
+              var linkHeader = new self.linkHeaderParser(jqXHR.getResponseHeader("Link"));
+              var lastPage = linkHeader.getLastPageNum();
+              // check if itemsPerPage should be updated to returned value
+              if (linkHeader.getPageSize() !== self.settings.itemsPerPage) {
+                self.settings.itemsPerPage = linkHeader.getPageSize();
+              }
+              // add pagination bar
+              $parent.append(self.getPaginationBar(lastPage * self.settings.itemsPerPage,
+                $currentContainer.attr("id")));
+            }
+          },
+          error: function() {
+            console.log("Could not load promotional content. AD-02");
+          }
+        });
+      }
+    },
+
+    /**
+     Retrieves and inserts a single promotion onto the page in the calling container. Will pass 
+     the host, absolute URI path, and additional parameters for the publish target, and promo ID. 
+     With these parameters, a post request will be formed that will retrieve a promo with an impressions ID.
+     */
+    singlePromo: function() {
+      var self = this;
+      var $container = $($(self)[0].element);
+      var url = self.settings.adsUrl;
+      var newsroomurl = self.settings.newsroomUrl;
```

Comment:
What do we use this for? We should probably remove this

Reply:
I use it on line 2070 to create the url of the default promo.